src/main/java/com/coastcapitalsavings/mvc/services/RequestService.java

```
@@ -300,4 +286,93 @@ private void sendRequestUpdatedEmail(Request r) {
         new Thread(runner).start();
     }
 
+    /**
+     * Updates the request status code to 'complete' as a user if the current status code is 'installed' and
+     * returns the updated request.
+     * @param userId            Id of the employee
+     * @param reqId             Id of the request to be updated
+     * @param oldStatusCode     Current status code of the request
+     * @param newStatusCode     New status code of the request
+     * @return
+     */
+    private Request userUpdateStatus(String userId, long reqId, String oldStatusCode, String newStatusCode) {
+        Request request = requestRepo.getRequestById(reqId);
+        if (oldStatusCode.equals("DENY"))
+            throw new DataIntegrityViolationException("RESOURCE_CANNOT_BE_MODIFIED");
+        if (newStatusCode.equals(oldStatusCode))
+            return request;  // nothing to be updated
+        if (newStatusCode.equals("COMP") && oldStatusCode.equals("INST")) {
+            requestRepo.putRequestStatusByApprover(reqId, userId, newStatusCode);
+            return updateRequestStatus(reqId, newStatusCode, userId);
+        }
+        return request;
```

Should throw an exception and return some status code along the lines of bad request for any other situation, instead of returning the request